CONTRIBUTING.md

```diff
@@ -1,63 +1,124 @@
-# Dotty Developer Guidelines
+# Thank you for contributing to Dotty
 
-These guidelines are meant to be a living document that should be changed and adapted as needed. We encourage changes that make it easier to achieve our goals in an efficient way.
```

Comment:
I think (not strongly) we should keep this paragraph since I feel it's still accurate.